src/core/model/table/column_layout_typed_relation_data.cpp

```diff
@@ -23,32 +18,20 @@ std::unique_ptr<ColumnLayoutTypedRelationData> ColumnLayoutTypedRelationData::Cr
     while (data_stream.HasNextRow()) {
         row = data_stream.GetNextRow();
 
-        if (row.empty() && num_columns == 1) {
-            row.emplace_back("");
-        } else if ((int)row.size() != num_columns) {
-            LOG(WARNING) << "Skipping incomplete rows";
+        if (row.size() != num_columns) {
+            LOG(WARNING) << "Unexpected number of columns for a row, skipping (expected "
+                         << num_columns << ", got " << row.size() << ")";
             continue;
         }
 
-        if (max_rows <= 0 || row_num < max_rows) {
-            int index = 0;
-            for (std::string& field : row) {
-                columns[index].push_back(std::move(field));
-                index++;
-                if (index >= num_columns) {
-                    break;
-                }
-            }
-        } else {
-            LOG(WARNING) << "Processed " << row_num << " rows and " << max_rows - row_num
-                         << " rows remain unprocessed due to `max_rows` parameter.";
-            break;
+        for (size_t index = 0; index < row.size(); ++index) {
+            std::string& field = row[index];
+            columns[index].push_back(std::move(field));
```

Comment:
Why not range-based for?

Reply:
Forgot to change.

Reply:
No, hold on, `columns[index]` uses `index`, I can't use a range-based `for` here.